yt/config.py

```diff
@@ -66,74 +76,132 @@
+        if metadata is None:
+            metadata = {}
+        self.config_root.update(new_values, metadata)
+
+    def has_section(self, section):
+        try:
+            self.config_root.get_child(section)
+            return True
+        except KeyError:
+            return False
+
+    def add_section(self, section):
+        self.config_root.add_child(section)
+
+    def remove_section(self, section):
+        if self.has_section(section):
+            self.config_root.remove_child(section)
+
+    def set(self, *args, metadata=None):
+        section, *keys, value = args
+        if metadata is None:
+            metadata = {"source": "runtime"}
+        self.config_root.upsert_from_list(
+            [section] + list(keys), value, extraData=metadata
+        )
+
+    def __setitem__(self, keys, value):
+        self.set(*keys, value, metadata=None)
 
     def __getitem__(self, key):
-        self.get(key[0], key[1])
+        section, *keys = key
+        return self.get(section, *keys)
+
+    def remove(self, *args):
+        self.config_root.pop_leaf(args)
+
+    def read(self, file_names):
+        if not isinstance(file_names, (tuple, list)):
+            file_names = (file_names,)
+
+        file_names_read = []
+        for fname in file_names:
+            if not os.path.exists(fname):
+                continue
+            metadata = {"source": f"file: {fname}"}
+            self.update(toml.load(fname), metadata=metadata)
+            file_names_read.append(fname)
+
+        return file_names_read
+
+    def write(self, fd):
+        value = self.config_root.as_dict()
+        config_as_str = toml.dumps(value)
 
-    def get(self, section, option, *args, **kwargs):
-        val = super(YTConfigParser, self).get(section, option, *args, **kwargs)
-        return os.path.expanduser(os.path.expandvars(val))
+        try:
+            fd.write(config_as_str)
+        except AttributeError:
+            with open(fd, mode="w") as fdd:
+                fdd.write(config_as_str)
 
 
-ytcfg = YTConfigParser(ytcfg_defaults)
-ytcfg.read([_OLD_CONFIG_FILE, CURRENT_CONFIG_FILE, "yt.cfg"])
-if not ytcfg.has_section("yt"):
-    ytcfg.add_section("yt")
+# Walk the tree up until we find a config file
+ytcfg = YTConfig()
+ytcfg.update(ytcfg_defaults, metadata={"source": "defaults"})
 
-# Now we have parsed the config file.  Overrides come from the command line.
+if os.path.exists(GLOBAL_CONFIG_FILE):
+    ytcfg.read(GLOBAL_CONFIG_FILE)
 
-# This should be implemented at some point.  The idea would be to have a set of
-# command line options, fed through an option parser, that would override
-# the settings in ytcfg.  *However*, because we want to have the command-line
-# scripts work, we'd probably want to have them only be long options, and also
-# along the lines of --yt-something=somethingelse.  The command line scripts
-# would then not get their options from sys.argv, but instead from this module.
+cwd = Path.cwd().resolve()
```

Comment:
I don't think calling `.resolve()` has any effect here
```suggestion
cwd = Path.cwd()
```